api/tests/test_auth_helpers.py

```
@@ -0,0 +1,110 @@
+        "iat": int(now.timestamp()),
+        "exp": int((now + exp_delta).timestamp()),
+        "last_activity": (now + last_activity_delta).isoformat(),
+    }
+    return pyjwt.encode(payload, get_settings().SECRET_KEY, algorithm=JWT_ALGORITHM)
+
+
+def test_jwt_absolute_expiry_reported_as_expired():
+    token = _make_token(exp_delta=timedelta(seconds=-10), last_activity_delta=timedelta(0))
+    payload, reason = decode_jwt_detailed(token)
+    assert reason == "expired"
+    assert payload is not None and payload["sub"] == "user-123"
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)